tools/dissector/sample_dissector.cpp

```diff
@@ -1256,6 +1286,82 @@ namespace OpenDDS
       base_->init_ws_fields();
     }
 
+    //-----------------------------------------------------------------------
+
+    Sample_Fixed::Sample_Fixed(unsigned digits, unsigned scale)
+    :
+      Sample_Dissector(""),
+      digits_(digits),
+      scale_(scale)
+    {
+    }
+
+    void Sample_Fixed::init_ws_fields() {
+      add_protocol_field(FT_DOUBLE);
+    }
+
+    size_t Sample_Fixed::dissect_i(Wireshark_Bundle &params) {
+      // Based on FACE/Fixed.h
+      // Fixed_T could not be used here because it is a templated class
+      size_t len = (digits_ + 2) / 2;
+
+      if (!params.get_size_only) {
+        int hf = get_hf();
+        if (hf == -1) {
+          throw Sample_Dissector_Error(
+            get_ns()  + " is not a registered wireshark field."
+          );
+        }
+
+#ifdef ACE_HAS_CDR_FIXED
+        // Extract a ACE_CDR::Fixed, give it to WS as a double, and display
+        // it using Fixed.to_string().
+        FACE::Octet raw[len];
```

Comment:
The bounds of a C Array need to be known at compile time.  You can use DDS::OctetSeq or std::vector as runtime-sized continuous byte containers.

Reply:
Yea, I copied it from the Fixed_T template and even changed the name without thinking. I'm going to set it to the max size of 32.